### ISSUE TITLE
Add geofencing as an authentication option

### ISSUE THREAD

**Issue by tgp1994** [PhilippC/keepass2android#14]
With phones, we're getting more and more hardware to use for authentication. I have Android 6.0, and in the system, there are extra options available for unlocking your phone - using a picture of your face, using your phone's location, even a connected Bluetooth device. Would it be possible to add functionality where we can Quick unlock our database by using location?

**Comment by Opensourcecommunitydevelopment** [comment]
i do like suggest the opposite, locking the database by leaving geofence (home)

**Comment by tgp1994** [comment]
@Opensourcecommunitydevelopment Yup - it's unlocked when you're home, and locked when you're away.

**Comment by 4-FLOSS-Free-Libre-Open-Source-Software** [comment]
could be extended to leaving bluetooth or wifi area (home||car||office)

**Comment by mirh** [comment]
[Trust Agent API](https://nelenkov.blogspot.com/2014/12/dissecting-lollipops-smart-lock.html) is probably what you'd use for this.
